### PR TITLE
[dns] Allow to specify query Class

### PR DESCRIPTION
**Pull request by Annih**
**Context**
In some cases we might want to probe other DNS class than the INET one; for instance we could probe the server version often exposed in a `TXT CH version.bind` record.

This PR offers the ability to do so by setting `query_class` in your prober configuration, the default is `"IN"` as it was before.

**Implementation note**
Most of the code is a copy/paste/replace of the `query_type` setting; except that the `setQuestion` call has been replaced by its "[slightly more verbose, but more flexible](https://godoc.org/github.com/miekg/dns)" alternative.

### CODE REVIEW

File: example.yml [hunk 1]
```diff
@@ -129,6 +129,15 @@ modules:
     dns:
       query_name: "prometheus.io"
       query_type: "SOA"
+  dns_bind_version:
```
Comment (brian-brazil) [inline example.yml:132]: This isn't really testing anything about the version. This is a niche enough use case that we can probably do without an example in the config.

File: prober/dns.go [hunk 2]
```diff
@@ -141,6 +141,16 @@ func ProbeDNS(ctx context.Context, target string, module config.Module, registry
 	registry.MustRegister(probeDNSAuthorityRRSGauge)
 	registry.MustRegister(probeDNSAdditionalRRSGauge)
 
+	qc := uint16(dns.ClassINET)
+	if module.DNS.QueryClass != "" {
+		var ok bool
+		qc, ok = dns.StringToClass[module.DNS.QueryClass]
+		if !ok {
+			level.Error(logger).Log("msg", "Invalid query class", "Class seen", module.DNS.QueryClass, "Existing classes", dns.ClassToString)
```
Comment (brian-brazil) [inline prober/dns.go:149]: This should be done in the config so that it can be caught when the config is loaded.

File: CONFIGURATION.md [hunk 3]
```diff
@@ -153,6 +153,8 @@ tls_config:
 
 [ transport_protocol: <string> | default = "udp" ] # udp, tcp
 
+[ query_class: <string> | default = "IN" ]
```
Comment (brian-brazil) [inline CONFIGURATION.md:156]: I'd put this below the type, as it'll be less common.

File: prober/dns.go (hunk 2 above)
Comment (Annih) [reply prober/dns.go:149]: Would that mean the [QueryType check L148](https://github.com/prometheus/blackbox_exporter/pull/635/files/1f04ae59feacb65cd358ff52a665b9e2da193dc6#diff-26673a9bcc326ad193c4d9832bfc4798L148) (should be also done in the config?
Would you accept a 2nd PR to move both checks outside this file?

File: example.yml (hunk 1 above)
Comment (Annih) [reply example.yml]: Removed :)

File: CONFIGURATION.md (hunk 3 above)
Comment (Annih) [reply CONFIGURATION.md]: Moved :)

File: prober/dns.go (hunk 2 above)
Comment (brian-brazil) [reply prober/dns.go:149]: Indeed, it should be. It's a small enough change it can all be done here.

**Comment by brian-brazil** [comment]
Did you get a chance to make that final change? I'd like to get a release out soon and would like to include this PR.

File: prober/dns.go (hunk 2 above)
Comment (Annih) [reply prober/dns.go:149]: done, sorry for the delay.
I added config checks but also kept the existing code in the prober.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**